### PR TITLE
doc: fix app extension build-args and improve qemu docs

### PR DESCRIPTION
**Pull request by ghost**
Application extension:
* The build-args property was not respected when building a west command.

Split off Qemu Networking fixes to #77852 as requested by @kartben 

See the result of the fix here: https://builds.zephyrproject.io/zephyr/pr/77601/docs/connectivity/networking/qemu_setup.html#network-connection-between-two-qemu-vms

### CODE REVIEW

**Comment by kartben** [comment]
This is touching two very different areas, would probably make sense to split into two PRs if possible. Thanks!

**Review by gmarull — DISMISSED** [review]

**Comment by fgrandel** [comment]
> This is touching two very different areas, would probably make sense to split into two PRs if possible. Thanks!

Will do.

**Comment by fgrandel** [comment]
> This is touching two very different areas, would probably make sense to split into two PRs if possible. Thanks!

@kartben  Done. Please re-review.

**Review by kartben — DISMISSED** [review]

File: doc/_extensions/zephyr/application.py [hunk 1]
```diff
@@ -320,8 +318,18 @@ def _generate_west(self, **kwargs):
...
+                board,
+                build_args,
+                west_args,
+                snippet_args,
+                shield_args,
+                build_dst,
+                src,
+                cmake_args,
+            )
+        )
```
Comment (pdgendt) [inline doc/_extensions/zephyr/application.py:332]: Not sure what the recommended way is, but I tend to prefer formatted string literals.

```suggestion
        content.append(
            f"west build -b {board}{build_args}{west_args}{snippet_args}"
            f"{shield_args}{build_dist}{src}{cmake_args}"
        )
```

Comment (fgrandel) [reply doc/_extensions/zephyr/application.py]: Sure, why not. :-) Fixed.

**Review by pdgendt — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Comment by fgrandel** [comment]
@gmarull Could you please re-approve - no changes, just split-off part of the PR.

**Review by gmarull — APPROVED** [review]

**Pull request merged**